drape_frontend/drape_engine.cpp

```diff
@@ -50,9 +50,14 @@ DrapeEngine::DrapeEngine(Params && params)
   m_textureManager = make_unique_dp<dp::TextureManager>();
   m_threadCommutator = make_unique_dp<ThreadsCommutator>();
 
-  int modeValue = 0;
-  if (!Settings::Get(LocationStateMode, modeValue))
-    modeValue = location::MODE_FOLLOW;
+  location::EMyPositionMode mode = params.m_initialMyPositionMode.first;
+  if (!params.m_initialMyPositionMode.second)
+  {
+    int modeValue;
+    if (!Settings::Get(LocationStateMode, modeValue))
```

Comment:
Does it make sense to add enum serialization into Settings, like we did for measurement?